### PR TITLE
feat(TCOMP-2475): Migration from higher version not protected for a connector

### PR DESCRIPTION
**Pull request by undx**
https://jira.talendforge.org/browse/TCOMP-2475

### CODE REVIEW

File: component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java [hunk 1]
```diff
@@ -566,9 +566,15 @@ public Map<String, String> migrate(final String id, final int version, final Map
                         .status(Status.NOT_FOUND)
                         .entity(new ErrorPayload(COMPONENT_MISSING, "Didn't find component " + id))
                         .build()));
-        if (version > comp.getVersion() || version == comp.getVersion()) {
+        if (version > comp.getVersion()) {
```
Comment (ozhelezniak-talend) [inline component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java:569]: Why not >=?
If the config version == the version, we shouldn't migrate it, right?

Comment (ozhelezniak-talend) [reply component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java:569]: Why do we call migration if versions ==

**Review by ozhelezniak-talend — COMMENTED** [review]
I have a question about the call of migration when the versions are equal.
And also I don't see a protection when the incoming version is higher than the one that is registered in component-runtime.
And I thought one of the points is to have some notification about it. Warning at least. 
Maybe I'm missing a point.

File: component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java (hunk 1 above)
Comment (undx) [reply component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java:569]: There's a comment on jira issue concerning studio's migration handling.
Migration fails in that case and the job cannot be opened. 
Same behavior observed in slack topic before the revert.

**Comment by undx** [comment]
> I have a question about the call of migration when the versions are equal. And also I don't see a protection when the incoming version is higher than the one that is registered in component-runtime. And I thought one of the points is to have some notification about it. Warning at least. Maybe I'm missing a point.

Answered to equals in comment.
There's a protection and warning preventing component migration call from higher version here
https://github.com/Talend/component-runtime/pull/898/files/4c91f86d3b6a5fdadda74bc4bb3a49450bbd7b98#diff-c897c2077dde44ed22410fa091a71724ed24211cfde6dd20348e11bf49a4d30aR570-R571

**Comment by ozhelezniak-talend** [comment]
> > I have a question about the call of migration when the versions are equal. And also I don't see a protection when the incoming version is higher than the one that is registered in component-runtime. And I thought one of the points is to have some notification about it. Warning at least. Maybe I'm missing a point.
> 
> Answered to equals in comment. There's a protection and warning preventing component migration call from higher version here https://github.com/Talend/component-runtime/pull/898/files/4c91f86d3b6a5fdadda74bc4bb3a49450bbd7b98#diff-c897c2077dde44ed22410fa091a71724ed24211cfde6dd20348e11bf49a4d30aR570-R571

So it will a warning...
Oh, don't know about that
It's really looks like smth is fishy is going on when someone tries to run the configuration that is higher than is known (registered in) by component-runtime

File: component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java (hunk 1 above)
Comment (ozhelezniak-talend) [reply component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java:569]: Oh my...
Okay

**Comment by undx** [comment]
> > > I have a question about the call of migration when the versions are equal. And also I don't see a protection when the incoming version is higher than the one that is registered in component-runtime. And I thought one of the points is to have some notification about it. Warning at least. Maybe I'm missing a point.
> > 
> > 
> > Answered to equals in comment. There's a protection and warning preventing component migration call from higher version here https://github.com/Talend/component-runtime/pull/898/files/4c91f86d3b6a5fdadda74bc4bb3a49450bbd7b98#diff-c897c2077dde44ed22410fa091a71724ed24211cfde6dd20348e11bf49a4d30aR570-R571
> 
> So it will a warning... Oh, don't know about that It's really looks like smth is fishy is going on when someone tries to run the configuration that is higher than is known (registered in) by component-runtime

Warning and call to migrate is skipped.

**Review by ozhelezniak-talend — APPROVED** [review]
LGTM

**Review by ozhelezniak-talend — APPROVED** [review]
lgtm

**Pull request merged**